titus-common/src/main/java/com/netflix/titus/common/util/time/TestWorldClock.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.common.util.time;
+
+import java.time.DayOfWeek;
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.concurrent.TimeUnit;
+
+class TestWorldClock implements TestClock {
+
+    private OffsetDateTime dateTime;
+
+    TestWorldClock(int year, Month month, int day) {
+        this.dateTime = OffsetDateTime.of(year, month.getValue(), day, 0, 0, 0, 0, ZoneOffset.UTC);
+    }
+
+    TestWorldClock() {
+        this.dateTime = OffsetDateTime.of(LocalDateTime.now(), ZoneOffset.UTC);
+    }
+
+    @Override
+    public long advanceTime(long interval, TimeUnit timeUnit) {
+        this.dateTime = dateTime.plusNanos(timeUnit.toNanos(interval));
```

Comment:
while this approach works fine most of the time, for large intervals, converting to nanos can overflow long.

Reply:
This is test clock, so as long we stay ~ year 2000 in our tests it should be fine.